packages/api/src/derives/cennzx/cennzxTrade.e2e.ts

```diff
@@ -0,0 +1,500 @@
+      const assetToSell = coreAssetId;
+      const assetToBuy = assetA;
+      const sellAmount = 50;
+      const expectedAssetPrice = await api.rpc.cennzx.sellPrice(assetToSell, sellAmount, assetToBuy);
+      const minSale = 1;
+      await api.tx.cennzx
+        .sellAsset(recipient, assetToSell, assetToBuy, sellAmount, minSale)
+        .signAndSend(alice, async ({ events, status }) => {
+          if (status.isFinalized && events !== undefined) {
+            for (const { event } of events) {
+              if (event.method === 'AssetPurchase') {
+                // check if ExtrinsicFailed or successful
+                const sellValue = event.data[4];
+                expect(sellValue.eq(expectedAssetPrice)).toBeTruthy();
+                done();
+              }
+            }
+          }
+        });
+    });
+
+    it('Get core asset from seller and transfer trade asset to charlie for exact trade asset amount', async done => {
+      const recipient = charlie.address; // in which case the bob receives the exchanged amount
+      const assetToSell = coreAssetId;
+      const assetToBuy = assetA;
+      const sellAmount = 50;
+      const expectedPrice = await api.rpc.cennzx.sellPrice(assetToSell, sellAmount, assetToBuy);
+      const minSale = 1;
+      const recipientBalaneBefore = await api.query.genericAsset.freeBalance(assetToBuy, recipient);
+      await api.tx.cennzx
+        .sellAsset(recipient, assetToSell, assetToBuy, sellAmount, minSale)
+        .signAndSend(alice, async ({ events, status }) => {
+          if (status.isFinalized && events !== undefined) {
+            for (const { event } of events) {
+              if (event.method === 'AssetPurchase') {
+                // check if ExtrinsicFailed or successful
+                const sellValue = event.data[4];
+                expect(sellValue.eq(expectedPrice)).toBeTruthy();
+                const recipientBalaneAfter = await api.query.genericAsset.freeBalance(assetToBuy, recipient);
+                expect(
+                  new BN(recipientBalaneBefore).add(new BN(sellValue)).eq(new BN(recipientBalaneAfter))
+                ).toBeTruthy();
+                done();
+              }
+            }
+          }
+        });
+    });
+
+    it('Can trade from asset to core for exact trade asset amount', async done => {
+      const recipient = null; // in which case the caller receives the exchanged amount
+      const assetToSell = assetA;
+      const assetToBuy = coreAssetId;
+      const sellAmount = 50;
+      const expectedCorePrice = await api.rpc.cennzx.sellPrice(assetToSell, sellAmount, assetToBuy);
+      const minSale = 1;
+      await api.tx.cennzx
+        .sellAsset(recipient, assetToSell, assetToBuy, sellAmount, minSale)
+        .signAndSend(alice, async ({ events, status }) => {
+          if (status.isFinalized && events !== undefined) {
+            let trade = false;
```

Comment:
what's the purpose of this variable `let trade`?

Reply:
don't need.. vll remove..